bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -241,6 +242,56 @@ impl From<UploadSource> for AttachmentSource {
     }
 }
 
+/// This type represents the "send progress" of a local event timeline item.
+#[derive(Clone, Copy, uniffi::Enum)]
+pub enum EventSendProgress {
+    /// A media is being uploaded.
+    MediaUpload {
```

Comment:
Since this isn't serialized: maybe we could make this enum with one variant a plain struct, instead? `MediaUploadProgress` or something like that?